### PR TITLE
Simplify build plugin license handling

### PR DESCRIPTION
**Pull request by breskeby**
- Use file property and conventions to avoid afterEvaluate hook
- Simplify root build script
- One little step closer to configuration cache compliance

### CODE REVIEW

File: build-tools-internal/src/integTest/java/org/elasticsearch/gradle/internal/test/TestClasspathUtils.java [hunk 1]
```diff
@@ -10,6 +10,12 @@
 
 import java.io.File;
 import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
```
Comment (pugnascotia) [inline build-tools-internal/src/integTest/java/org/elasticsearch/gradle/internal/test/TestClasspathUtils.java:18]: I don't understand why have lots of new import but no other code changes in this file?

File: build-tools/src/testFixtures/java/org/elasticsearch/gradle/internal/test/GradleIntegrationTestCase.java [hunk 2]
```diff
@@ -176,6 +177,9 @@ protected void assertNoDeprecationWarning(BuildResult result) {
 
     protected void assertBuildFileExists(BuildResult result, String projectName, String path) {
         Path absPath = getBuildDir(projectName).toPath().resolve(path);
+        for (File file : absPath.getParent().toFile().listFiles()) {
+            System.out.println("file = " + file);
```
Comment (pugnascotia) [inline build-tools/src/testFixtures/java/org/elasticsearch/gradle/internal/test/GradleIntegrationTestCase.java:181]: Is this some leftover debugging code?

File: build-conventions/src/main/java/org/elasticsearch/gradle/internal/conventions/LicensingPlugin.java [hunk 3]
```diff
@@ -34,24 +33,23 @@ public LicensingPlugin(ProviderFactory providerFactory) {
...
                 )
         );
+
         // Default to the SSPL+Elastic dual license
-        project.getExtensions().getExtraProperties().set("licenseCommit", licenseCommitProvider);
-        project.getExtensions().getExtraProperties().set("projectLicenses", convention);
+        project.getExtensions().getExtraProperties().set("projectLicenses", licensesProperty);
     }
 
     private boolean isSnapshotVersion(Project project) {
```
Comment (mark-vieira) [inline build-conventions/src/main/java/org/elasticsearch/gradle/internal/conventions/LicensingPlugin.java:55]: Can we ditch this and just use `BuildParams.isSnapshot`?

**Review by mark-vieira — APPROVED** [review]
Couple minor comments but otherwise LGTM.

File: build-tools-internal/src/integTest/groovy/org/elasticsearch/gradle/internal/PublishPluginFuncTest.groovy [hunk 4]
```diff
@@ -277,8 +277,8 @@ class PublishPluginFuncTest extends AbstractGradleFuncTest {
                     
             // requires elasticsearch artifact available
             tasks.named('bundlePlugin').configure { enabled = false }
-            licenseFile = file('license.txt')
-            noticeFile = file('notice.txt')
+            licenseFile.set(file('license.txt'))
```
[the remark this answers is not in the document]
Comment (breskeby) [reply build-tools-internal/src/integTest/groovy/org/elasticsearch/gradle/internal/PublishPluginFuncTest.groovy:280]: unfortunately not

File: build-tools-internal/src/integTest/java/org/elasticsearch/gradle/internal/test/TestClasspathUtils.java (hunk 1 above)
Comment (breskeby) [reply build-tools-internal/src/integTest/java/org/elasticsearch/gradle/internal/test/TestClasspathUtils.java:18]: fixed.

File: build-tools/src/testFixtures/java/org/elasticsearch/gradle/internal/test/GradleIntegrationTestCase.java (hunk 2 above)
Comment (breskeby) [reply build-tools/src/testFixtures/java/org/elasticsearch/gradle/internal/test/GradleIntegrationTestCase.java:181]: good catch. removed.

File: x-pack/build.gradle [hunk 5]
```diff
@@ -40,6 +40,8 @@ subprojects {
     ext.projectLicenses.set(['Elastic License 2.0': ext.elasticLicenseUrl.get()])
   }
 
-  project.ext.licenseFile = rootProject.file('licenses/ELASTIC-LICENSE-2.0.txt')
-  project.ext.noticeFile = xpackRootProject.file('NOTICE.txt')
+  project.getPluginManager().withPlugin("elasticsearch.build") {
```
[the remark this answers is not in the document]
Comment (breskeby) [reply x-pack/build.gradle:43]: fixed

File: build-conventions/src/main/java/org/elasticsearch/gradle/internal/conventions/LicensingPlugin.java (hunk 3 above)
Comment (breskeby) [reply build-conventions/src/main/java/org/elasticsearch/gradle/internal/conventions/LicensingPlugin.java:55]: BuildParams is build-tools-internal, this is build-conventions. so unfortunately we cannot

**Pull request merged**